samples/benchmarks/coremark/README.rst

```diff
@@ -0,0 +1,155 @@
+Enable ``COREMARK_MODE_FLASH_AND_RUN`` if you want the sample to run for the first time upon startup.
+
+User interface
+**************
+
+Each target CPU has an assigned button and LED that are responsible for triggering the benchmark's start and ``test in progress``:
+
+Button 1:
+   Start the benchmark run on the Application Core.
+Button 2:
+   Start the benchmark run on the Network/Radio Core.
+Button 3:
+   Start the benchmark run on the PPR Core.
+LED 1:
+   Indicates ``test in progress`` on the Application Core.
+LED 2:
+   Indicates ``test in progress`` on the Network/Radio Core.
+LED 3:
+   Indicates ``test in progress`` on the PPR Core.
+
+Configuration
+*************
+
+CoreMark runs tests multiple times. The number of iterations is defined with the ``CONFIG_COREMARK_ITERATIONS`` option.
+By default, the iteration quantity is set to the minimum time required for valid results.
+For valid results, the test should be executed for at least 10 seconds.
+
+CoreMark has three predefined run types, with the predefined data sizes, that are used for data algorithms:
+
+* ``CONFIG_COREMARK_RUN_TYPE_PERFORMANCE`` - defines ``CONFIG_COREMARK_DATA_SIZE`` to 2000 bytes.
+* ``CONFIG_COREMARK_RUN_TYPE_PROFILE`` - defines ``CONFIG_COREMARK_DATA_SIZE`` to 1200 bytes.
+* ``CONFIG_COREMARK_RUN_TYPE_VALIDATION`` - defines ``CONFIG_COREMARK_DATA_SIZE`` to 500 bytes.
+
+However, you can specify a custom ``CONFIG_COREMARK_DATA_SIZE`` value.
+If you want to register your results, see the Submitting Results chapter in the `CoreMark GitHub`_ repository.
+
+Additionally, the ``CONFIG_COREMARK_DATA_SIZE`` option requires available memory space.
+By default, the ``CONFIG_COREMARK_MEMORY_METHOD_STACK`` option is selected.
+In this case, the main thread stack is used for data allocation.
+To change the main stack size, use the ``CONFIG_MAIN_STACK_SIZE`` option.
+
+If you select the ``CONFIG_COREMARK_MEMORY_METHOD_MALLOC`` option, Zephyr :c:func:`k_malloc\k_free` functions will be used to allocate the memory in the heap.
+Adjust the heap size with the ``CONFIG_HEAP_MEM_POOL_SIZE`` config option.
+
+If you select the ``CONFIG_COREMARK_MEMORY_METHOD_STATIC`` option, data will be allocated to the RAM as a regular static variable.
+In this case you do not need to be aware of the main thread stack size or heap memory pool size.
+
+CoreMark can be executed in a multithread mode.
+To specify a number of threads, use the ``CONFIG_COREMARK_THREADS_NUMBER`` option.
+You can configure up to four threads.
+In the multithread mode, CoreMark is executed in all threads simultaneously.
+By default ``CONFIG_TIMESLICE_SIZE`` is set to 10 ms which imitates RTOS multithread usage.
+However, in the final report thread execution is combined.
+In multithread mode, the ``CONFIG_COREMARK_DATA_SIZE`` option is allocated for each thread separately.
+Only the ``CONFIG_COREMARK_MEMORY_METHOD_STACK`` and ``CONFIG_COREMARK_MEMORY_METHOD_MALLOC`` memory methods can be used with multiple threads.
```

Comment:
Asserts in kconfig - such a combination exists? Maybe we should add some dependency for COREMARK_MEMORY_METHOD_STATIC that COREMARK_THREADS_NUMBER equals to 1

Reply:
Assert in code - sorry for confusion

Reply:
Yes, it would be nice to add the mentioned dependency in kconfig. Alternatively, you can change COREMARK_THREADS_NUMBER range to "1 1" if CONFIG_COREMARK_MEMORY_METHOD_STATIC. I'm okay with both approaches.

Reply:
If we add proper dependencies in Kconfig assert in code may be not needed, but it will still be nice to have to double-check configuration.